DAOS-13073 build: make STACK_MMAP=1 the default for v2.3.108-tb

bfaccini opened the pull request:
To build and test all with mmap()'ed ULT stacks feature enabled.
Based on release/2.4 branch at tag v2.3.108-tb, with additional commits from DAOS-4183/PR-10808.


### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

Commit:
DAOS-4183 engine: reduce virtual memory and swap footprint

In order to limit the virtual memory and swap footprint, only
mmap() the exact requested stack size (to be rounded up to the page
size by the Kernel!), different sizes will be managed by a b-tree,
and MAP_NORESERVE flag will now be used.

Required-githooks: true

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

Commit:
DAOS-4183 engine: cosmetic changes to fix coding rules warns

All is in title !!...

Required-githooks: true

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

Commit:
DAOS-4183 engine: fix typo of parenthesis in excess

When pushing previous commit for cosmetic fixes, I have introduced
a compile error due to a parenthesis left in excess !!...

Required-githooks: true

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

Commit:
DAOS-4183 engine: more cosmetic fixes

Alignment, line > 100 chars, ...

Required-githooks: true

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

Commit:
DAOS-4183 engine: one more cosmetic fix

Had to remove a blank inserted within tabs in a continuation
line ... :-(

Required-githooks: true

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

jolivier23 committed:
DAOS-623 build: bump version to 2.3.108 (#12288)

REBASE AND LAND TO RELEASE/2.4 AFTER CREATION

Bump the version  to 2.3.108 now that the release branch has been
created.
Adjust Jenkinsfile accordingly.
Fix a few links in the documentation that point to master branch.

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

Commit:
DAOS-13797 build: make STACK_MMAP=1 the default

To build and test all with mmap()'ed ULT stacks
feature enabled.

Required-githooks: true

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage NLT on EL 8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-12788/1/testReport/

Commit:
DAOS-13037 nlt: bump max-log-size for STACK_MMAP=1

Seems that NLT logging needs a lightly bumped log
file size limit than 1.5GB when running with STACK_MMAP=1.

Required-githooks: true

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Unit Test on EL 8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-12788/2/testReport/

daosbuild1 commented:
Test stage NLT on EL 8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-12788/2/testReport/

Commit:
DAOS-13037 nlt: bump max-log-size for STACK_MMAP=1

Seems that NLT logging needs a lightly bumped log
file size limit than 1.5GB when running with STACK_MMAP=1.

Bur this time, really update option value on real command
line instead of in comment !!...

Required-githooks: true

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12788/3/execution/node/1088/log

daosbuild1 commented:
Test stage Functional on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12788/4/execution/node/384/log
